test/nlp/quasi-newton.jl

```diff
@@ -6,69 +6,71 @@
     J(x) = [1.0 -2.0; -0.5x[1] -2.0x[2]]
 
     for (QNM, QNO) in [(LSR1Model, LSR1Operator), (LBFGSModel, LBFGSOperator)]
-      nlp = QNM(SimpleNLPModel())
-      n = nlp.meta.nvar
-      m = nlp.meta.ncon
+      for dt in [Float64, Float32]
```

Comment:
To avoid changing the whole test, you can the `dt in [Float64, Float32]` at the end on the upper `for`. We can keep the indentation that way.
Also, I feel like `T` should be used instead of `dt`.